frb_codegen/src/ir/ty.rs

```diff
@@ -41,6 +41,24 @@ impl IrType {
             _ => "",
         }
     }
+
+    #[inline]
+    pub fn as_primitive(&self) -> Option<&IrTypePrimitive> {
+        match self {
+            Primitive(repr) | Delegate(IrTypeDelegate::PrimitiveEnum { repr, .. }) => Some(repr),
+            _ => None,
+        }
+    }
+
+    #[inline]
+    pub fn is_primitive(&self) -> bool {
+        self.as_primitive().is_some()
+    }
+
+    #[inline]
+    pub fn is_struct(&self) -> bool {
+        matches!(self, StructRef(_) | EnumRef(_))
+    }
```

Comment:
We could rename it to better reflect its nature, but I think for now it's fine since fieldless enums are always wrapped in a Delegate which is decently explicit. Part of the reason why they share the name EnumRef is that they have some shared code that needs some reorganizing the next time we refactor.